### PR TITLE
fix: Correct debug dir handling on Windows

### PR DESCRIPTION
**Pull request by llunak**
The drive letter removal was incorrect.

### CODE REVIEW

**Review by louiscaron — APPROVED** [review]

**Comment by louiscaron** [comment]
I had done a simular fix on my tricore support change but yours in much more in the "ccache" spirit. I will rebase as soon as yours gets merged.  In the meantime, could you allow the worklows on my PR: https://github.com/ccache/ccache/pull/942

File: src/util/path.cpp [hunk 1]
```diff
@@ -58,4 +58,15 @@ to_absolute_path(nonstd::string_view path)
   }
 }
 
+std::string
+to_absolute_path_no_drive(nonstd::string_view path)
```
Comment (jrosdahl) [inline src/util/path.cpp:62]: It would be great it you could add some tests of this to `unittest/test_util_path.cpp`.

Comment (llunak) [reply src/util/path.cpp:62]: Added.

**Comment by jrosdahl** [comment]
Thanks.

**Pull request merged**